cscs-checks/apps/amber/amber_check.py

```diff
@@ -57,103 +50,61 @@
+
+
+class AmberCheckCSCS(Amber_NVE):
     modules = ['Amber']
+    valid_prog_environs = ['builtin']
+    strict_check = True
     extra_resources = {
         'switches': {
             'num_switches': 1
         }
     }
     maintainers = ['VH', 'SO']
-    tags = {'scs', 'external-resources'}
-
-    benchmark = parameter([
-        # NVE simulations
-        'Cellulose_production_NVE',
-        'FactorIX_production_NVE',
-        'JAC_production_NVE_4fs',
-        'JAC_production_NVE',
-    ])
-
-    @run_after('init')
-    def download_files(self):
-        self.prerun_cmds = [
-            # cannot use wget because it is not installed on eiger
-            f'curl -LJO https://github.com/victorusu/amber_benchmark_suite'
-            f'/raw/main/amber_16_benchmark_suite/PME/{self.benchmark}.tar.bz2',
-            f'tar xf {self.benchmark}.tar.bz2'
-        ]
-
-    @run_after('init')
-    def set_energy_and_tolerance_reference(self):
-        self.ener_ref = {
-            # every system has a different reference energy and drift
-            'Cellulose_production_NVE': (-443246, 5.0E-05),
-            'FactorIX_production_NVE': (-234188, 1.0E-04),
-            'JAC_production_NVE_4fs': (-44810, 1.0E-03),
-            'JAC_production_NVE': (-58138, 5.0E-04),
-        }
-
-    @run_after('setup')
-    def set_executable_opts(self):
-        self.executable_opts = ['-O',
-                                '-i', self.input_file,
-                                '-o', self.output_file]
-        self.keep_files = [self.output_file]
-
-    @run_after('setup')
-    def set_sanity_patterns(self):
-        energy = sn.extractsingle(r' Etot\s+=\s+(?P<energy>\S+)',
-                                  self.output_file, 'energy', float, item=-2)
-        energy_reference = self.ener_ref[self.benchmark][0]
-        energy_diff = sn.abs(energy - energy_reference)
-        ref_ener_diff = sn.abs(self.ener_ref[self.benchmark][0] *
-                               self.ener_ref[self.benchmark][1])
-        self.sanity_patterns = sn.all([
-            sn.assert_found(r'Final Performance Info:', self.output_file),
-            sn.assert_lt(energy_diff, ref_ener_diff)
-        ])
-
-    @run_after('setup')
-    def set_generic_perf_references(self):
-        self.reference.update({'*': {
-            self.benchmark: (0, None, None, 'ns/day')
-        }})
-
-    @run_after('setup')
-    def set_perf_patterns(self):
-        self.perf_patterns = {
-            self.benchmark: sn.extractsingle(r'ns/day =\s+(?P<perf>\S+)',
-                                             self.output_file, 'perf',
-                                             float, item=1)
-        }
 
+    @run_before('run')
+    def set_perf_reference(self):
+        for key in list(self.REFERENCE_DICT):
+            if self.current_partition.fullname in key:
+                d = {self.current_partition.fullname:
+                     self.REFERENCE_DICT[key]}
+                self.reference = d
```

Comment:
I would say that we make this `set_perf_reference` a different hook for both the GPU and CPU cases, simply because their references above are compacted in a different format. So move this hook into the GPU-specific class, and then create a similar one in the GPU class where you also deal with the `small` & `large` scopes of the dictionary.

This also avoids having to store another reference to this `REFERENCE_DICT` in the class, which is a bit confusing.
```suggestion
        for key, val in REFERENCE_GPU_PERFORMANCE.items():
            if self.current_partition.fullname in key:
                self.reference = {'*': val}
                break
        else:
            raise ValueError(
                f'could not find a reference for the current '
                f'partition {self.current_partition.fullname!r}'
            )
```